lib/python/rose/config.py

```diff
@@ -1118,26 +1120,39 @@ def _string_node_dump(self, key, node, handle, env_escape_ok):
 
         """
         state = node.state
-        values = node.value.split("\n")
+        try:
+            values = node.value.decode().split("\n")
+        except AttributeError:
+            values = node.value.split("\n")
         for comment in node.comments:
-            handle.write(self._comment_format(comment))
+            self._write_safely(handle, self._comment_format(comment))
         value0 = values.pop(0)
         if env_escape_ok:
             value0 = env_var_escape(value0)
-        handle.write(state + key + self.char_assign + value0)
-        handle.write("\n")
+        self._write_safely(handle, state + key + self.char_assign + value0)
+        self._write_safely(handle, "\n")
         if values:
             indent = " " * len(state + key)
             for value in values:
                 if env_escape_ok:
                     value = env_var_escape(value)
-                handle.write(indent + self.char_assign + value + "\n")
+                self._write_safely(handle,
+                                   indent + self.char_assign + value + "\n")
 
     @classmethod
     def _comment_format(cls, comment):
         """Return text representation of a configuration comment."""
         return "#%s\n" % (comment)
 
+    @staticmethod
```

Comment:
I suggest that I replace this with the code in my gist at https://gist.github.com/wxtim/cc3f4e653d37616ff24714bfbc32444b.